src/pages/interviews.tsx

```diff
@@ -145,18 +163,23 @@ function Applicants({ type, applicants, interviews, refetchInterviews,
 }
 
 function Applicant({ type, applicant, interviews, refetchInterviews,
-  refetchApplicants
+  refetchApplicants, updateSource
  } : {
   type: InterviewType,
   applicant: User,
   interviews: Interview[],
   refetchInterviews: () => void,
   refetchApplicants: () => void,
+  updateSource:(id: string, source: string) => void,
```

Comment:
this function should not need parameter `id`

Reply:
How do you get the source of each applicant to sort?` const comp = (sources[a1.id] || "").localeCompare(sources[a2.id]);`
You don't have source attribute on applicant.
Do you need to call  trpcNext.users.getApplicant.useQuery to get the source?
```
const source = (data?.application as Record<string, any> | null)
    ?.[menteeSourceField];
```

Reply:
The caller already knows the id of the applicant. At line 153, you simply do `updateSource={source => updateSource(a.id, source)}`